docs/monitor-kubernetes.md

```diff
@@ -0,0 +1,305 @@
+
+2. [Helm v3](https://docs.helm.sh/using_helm/#installing-helm).
+
+## Procedure
+
+### Set up authentication in PMM Server
+
+To access the PMM Server resources and perform actions on the server, configure authentication.
+
+1. Get the PMM API key. The key must have the role "Admin".
+
+    === "From PMM UI" 
+
+        [Generate the PMM API key](https://docs.percona.com/percona-monitoring-and-management/details/api.html#api-keys-and-authentication){.md-button} 
+
+    === "From command line"
+
+        You can query your PMM Server installation for the API
+        Key using `curl` and `jq` utilities. Replace `<login>:<password>@<server_host>` placeholders with your real PMM Server login, password, and hostname in the following command:
+
+        ``` {.bash data-prompt="$" }
+        $ API_KEY=$(curl --insecure -X POST -H "Content-Type: application/json" -d '{"name":"operator", "role": "Admin"}' "https://<login>:<password>@<server_host>/graph/api/auth/keys" | jq .key)
+        ```
+
+    !!! note
+
+        The API key is not rotated. 
+
+2. Encode the API key with base64.
+
+    === "in Linux" 
+
+        ````{.bash data-prompt="$" }
+        $ echo -n <API-key> | base64 --wrap=0
+        ````
+
+    === "in macOS"   
+        ```{.bash data-prompt="$" }
+        $ echo -n <API-key> | base64 
+        ```
+
+3. Create the YAML file for the [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) and specify the API key value within. Let's name this file `pmm-api-vmoperator.yaml`.
+
+    ```yaml title="pmm-api-vmoperator.yaml"
+    apiVersion: v1
+    data:
+      api_key: <base-64-encoded-API-key>
+    kind: Secret
+    metadata:
+      name: pmm-token-vmoperator
+      #namespace: default
+    type: Opaque
+    ```
+
+4. Create the Namespace where you want to set up monitoring. The following command creates the Namespace `monitoring-system`. You can specify a different name. In the latter steps, specify this name instead of the `<namespace>` placeholder.
+    
+    ```{.bash data-prompt="$" }
+    $ kubectl create namespace monitoring-system
+    ```
+
+5. Create the Secrets object using the YAML file you created previously. Replace the `<filename>` placeholder with your value.
+
+    ```{.bash data-prompt="$" }
+    $ kubectl apply -f pmm-api-vmoperator.yaml -n <namespace>
+    ```
+
+6. Check that the secret is created. The following command checks the secret for the resource named `pmm-token-vmoperator` (as defined in the `metadata.name` option in the secrets file). If you defined another resource name, specify your value.  
+
+   ```{.bash data-prompt="$" }
+   $ kubectl get secret pmm-token-vmoperator -n <namespace>
+   ```
+
+### Create a ConfigMap to mount for kube-state-metrics 
+
+The `kube-state-metrics` (KSM) is a simple service that listens to the Kubernetes API server and generates metrics about the state of various objects - Pods, Deployments, Services and Custom Resources. 
```

Comment:
As per the doc https://github.com/kubernetes/kube-state-metrics/blob/main/docs/cli-arguments.md , capturing custom resource metrics is still experimental, we need to mention this somewhere, maybe mention this whole document as Beta or something . @spron-in May share more thoughts.
```
--custom-resource-state-config string        Inline Custom Resource State Metrics config YAML (experimental)
      --custom-resource-state-config-file string   Path to a Custom Resource State Metrics config file (experimental)
```